eZ/Publish/Core/Search/Common/Slot/SwapLocation.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common\Slot;
+
+use eZ\Publish\Core\SignalSlot\Signal;
+
+/**
+ * A Search Engine slot handling SwapLocationSignal.
+ */
+class SwapLocation extends AbstractSubtree
+{
+    /**
+     * Receive the given $signal and react on it.
+     *
+     * @param Signal $signal
+     */
+    public function receive(Signal $signal)
+    {
+        if (!$signal instanceof Signal\LocationService\SwapLocationSignal) {
+            return;
+        }
+
+        $this->indexSubtree($signal->location1Id);
+        $this->indexSubtree($signal->location2Id);
```

Comment:
Yes. I think [the test](https://github.com/alongosz/ezpublish-kernel/blob/bfa96cf3d175bb19f325bd8bff6f28a928690f4e/eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php#L312-L324) passes because for Solr Locations are nested in Content. Tried to figure out why it passes for Elasticsearch without re-indexing locations. Maybe because I'm swapping mainLocations.
Anyway, fixed in bfa96cf.